packages/two_dimensional_scrollables/lib/src/tree_view/tree.dart

```diff
@@ -0,0 +1,1056 @@
+  ///
+  /// Beyond root nodes, whether or not a given [TreeViewNode] is displayed
+  /// depends on the [TreeViewNode.isExpanded] value of its parent. The
+  /// [TreeView] will set the [TreeViewNode.parent] and [TreeViewNode.depth] as
+  /// nodes are built on demand to ensure the integrity of the tree.
+  final List<TreeViewNode<T>> tree;
+
+  /// Called to build an entry of the [TreeView] for the given [TreeViewNode].
+  ///
+  /// By default, if this is unset, the [TreeView.defaultTreeNodeBuilder] is
+  /// used.
+  final TreeViewNodeBuilder treeNodeBuilder;
+
+  /// Builds the [TreeRow] that describes the row for the provided
+  /// [TreeViewNode].
+  ///
+  /// By default, if this is unset, the [TreeView.defaultTreeRowBuilder]
+  /// is used.
+  final TreeViewRowBuilder treeRowBuilder;
+
+  /// If provided, the controller can be used to expand and collapse
+  /// [TreeViewNode]s, or lookup information about the current state of the
+  /// [TreeView].
+  final TreeViewController? controller;
+
+  /// Called when a [TreeViewNode] expands or collapses.
+  ///
+  /// This will not be called if a [TreeViewNode] does not have any children.
+  final TreeViewNodeCallback? onNodeToggle;
+
+  /// Whether to wrap each row of the tree in an [AutomaticKeepAlive].
+  ///
+  /// Typically, lazily laid out children are wrapped in [AutomaticKeepAlive]
+  /// widgets so that the children can use [KeepAliveNotification]s to preserve
+  /// their state when they would otherwise be garbage collected off-screen.
+  ///
+  /// This feature (and [addRepaintBoundaries]) must be disabled if the children
+  /// are going to manually maintain their [KeepAlive] state. It may also be
+  /// more efficient to disable this feature if it is known ahead of time that
+  /// none of the children will ever try to keep themselves alive.
+  ///
+  /// Defaults to true.
+  final bool addAutomaticKeepAlives;
+
+  /// Whether to wrap each row in a [RepaintBoundary].
+  ///
+  /// Typically, children in a scrolling container are wrapped in repaint
+  /// boundaries so that they do not need to be repainted as the list scrolls.
+  /// If the children are easy to repaint (e.g., solid color blocks or a short
+  /// snippet of text), it might be more efficient to not add a repaint boundary
+  /// and instead always repaint the children during scrolling.
+  ///
+  /// Defaults to true.
+  final bool addRepaintBoundaries;
+
+  /// Used to override the toggle animation's curve and duration.
+  ///
+  /// If [AnimationStyle.duration] is provided, it will be used to override
+  /// the [TreeView.defaultAnimationDuration], which is 150
+  /// milliseconds.
+  ///
+  /// If [AnimationStyle.curve] is provided, it will be used to override
+  /// the [TreeView.defaultAnimationCurve], which is [Curves.linear].
+  ///
+  /// To disable the tree animation, use [AnimationStyle.noAnimation].
+  final AnimationStyle? animationStyle;
```

Comment:
`animationStyle` could be named `toggleAnimationStyle`, which makes it easier to understand what type of animation this property customizes. In this case it used for toggle animation based on the docs. (Noticed this when reviewing examples)

E.g. [expansionAnimationStyle](https://main-api.flutter.dev/flutter/material/ExpansionTile/expansionAnimationStyle.html) is used to customize `ExpansionTile` expand/collapse animation.